test/core/end2end/invalid_call_argument_test.c

```diff
@@ -304,6 +304,11 @@ static void test_receive_initial_metadata_twice_at_client() {
   grpc_op *op;
   prepare_test(1);
   op = g_state.ops;
+  op->op = GRPC_OP_SEND_INITIAL_METADATA;
```

Comment:
I'd like to avoid this if possible.

What about adding deadline to grpc_call_element_args: the client would be able to configure the deadline at call-stack construction time, meaning it wouldn't need to receive a send initial metadata, meaning that this change would be unnecessary.